### PR TITLE
[PR/623 - pt 6] Refactor Task Executor Registration to Primarily Use Configurations for MachineDefinition

### PR DESCRIPTION
**Pull request by fdc-ntflx**
### Context

In this PR, we refine the task executor registration process to primarily fetch machine definitions from configurations (set by the control plane during ASG creation), rather than relying solely on system parameters. This approach offers greater accuracy as system parameters may not be entirely precise.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/config/WorkerConfiguration.java [hunk 1]
```diff
@@ -155,6 +155,18 @@ default Time getHeartbeatInterval() {
     @DefaultNull
     File getLocalStorageDir();
 
+    @Config("mantis.taskexecutor.hardware.cpu-cores")
```
Comment (Andyz26) [inline mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/config/WorkerConfiguration.java:158]: why are these moving to worker config vs the current reporting structure from a util/hardware layer? It doesn't seem intuitive to me these metadata should be part of the "configuration" object.

Comment (sundargates) [reply mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/config/WorkerConfiguration.java:158]: We have had bugs in the past where the machine would have a profile that does not match the requested resources resulting in scheduling issues on the control plane.  Hence, this change is expected to address that.

**Review by sundargates — APPROVED** [review]

**Review by Andyz26 — APPROVED** [review]

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java [hunk 2]
```diff
@@ -381,7 +381,7 @@ public boolean areSchedulingAttributeConstraintsSatisfied(SchedulingConstraints
             .stream()
             .collect(Collectors.toMap(entry -> entry.getKey().toLowerCase(), Map.Entry::getValue));
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java:382]: should the values use lower-case too?

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/JobDefinitionResolver.java [hunk 3]
```diff
@@ -76,8 +78,14 @@ JobDefinition getResolvedJobDefinition(final String user, final JobDefinition gi
...
-        // inherit labels from cluster if not specified
-        List<Label> labels = (resolvedJobDefn.getLabels() != null && !resolvedJobDefn.getLabels().isEmpty()) ? resolvedJobDefn.getLabels() : jobClusterMetadata.getJobClusterDefinition().getLabels();
+        // Inherit labels from cluster, if resolvedJobDefn has labels, override the existing ones
+        Map<String, Label> labelMap = jobClusterMetadata.getJobClusterDefinition().getLabels().stream()
+            .collect(Collectors.toMap(Label::getName, label -> label));
+        if (resolvedJobDefn.getLabels() != null && !resolvedJobDefn.getLabels().isEmpty()) {
+            resolvedJobDefn.getLabels()
+                .forEach(label -> labelMap.put(label.getName(), label));
+        }
+        List<Label> labels = new ArrayList<>(labelMap.values());
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/JobDefinitionResolver.java:88]: nit: immutable list?

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java [hunk 4]
```diff
@@ -1614,17 +1614,21 @@ private ScheduleRequest createSchedulingRequest(
...
          */
-        private Map<String, String> mergeJobDefAndArtifactAssigmentAttributes(ArtifactID artifactID) {
+        private Map<String, String> mergeJobDefAndArtifactAssigmentAttributes(String artifactName) {
+            LOGGER.info("[fdc-91] cluster getLabels() LABELS --> {}", jobClusterDefinition.getLabels());
+            LOGGER.info("[fdc-91] mantisJobMetaData.getJobDefinition() LABELS --> {}", mantisJobMetaData.getJobDefinition().getLabels());
             try {
-                JobArtifact artifact = jobStore.getJobArtifact(artifactID);
+                LOGGER.info("[fdc-91] mergeJobDefAndArtifactAssigmentAttributes() artifactName --> {}   ---> {}", artifactName, artifactName.substring(0, artifactName.indexOf('.')));
+
+                JobArtifact artifact = jobStore.getJobArtifact(ArtifactID.of(artifactName.substring(0, artifactName.indexOf('.'))));
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java:1626]: nit: move the artifact name handling to some static util functions.

File: mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/config/WorkerConfiguration.java [hunk 5]
```diff
@@ -170,6 +183,11 @@ default Map<String, String> getTaskExecutorAttributes() {
             return ImmutableMap.of();
         }
 
-        return Splitter.on(",").withKeyValueSeparator(':').split(input);
+        Map<String, String> attributes = Splitter.on(",").withKeyValueSeparator(':').split(input);
+
+        // filter out entries where the value matches the pattern "${.*}"
```
Comment (Andyz26) [inline mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/config/WorkerConfiguration.java:188]: can this be fixed on the input side?

File: mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/MachineDefinitionUtils.java [hunk 6]
```diff
@@ -17,8 +17,17 @@
...
+    /**
+     * Creates a MachineDefinition object using system parameters.
+     *
+     * @param workerPorts Worker ports object.
+     * @param networkBandwidthInMB Network bandwidth capacity in MB.
+     * @return MachineDefinition object.
+     */
     public static MachineDefinition sys(WorkerPorts workerPorts, double networkBandwidthInMB) {
         return new MachineDefinition(
             Hardware.getNumberCPUCores(),
```
Comment (sundargates) [inline mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/MachineDefinitionUtils.java:33]: Let's maybe get rid of this and just have the user specify the node they are running the TE on.

Comment (sundargates) [reply mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/MachineDefinitionUtils.java:33]: It appears like it's already not being used.

File: mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/config/WorkerConfiguration.java (hunk 5 above)
Comment (fdc-ntflx) [reply mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/config/WorkerConfiguration.java]: I looked into this but it's a bit tricky and worried to introduce some kinda of regression..

**Pull request closed without merging**